sycl/test/backend/cuda/interop_get_native.cpp

```diff
@@ -0,0 +1,90 @@
+// REQUIRES: cuda
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -I%opencl_include_dir -I%cuda_toolkit_include -o %t.out -lcuda -lsycl
+// RUN: env SYCL_DEVICE_TYPE=GPU %t.out
+
+//==---------- interop_get_native.cpp - SYCL cuda get_native tests ---------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <CL/sycl.hpp>
+#include <cuda.h>
+#include <iostream>
```

Comment:
Shouldn't SYCL/backend/cuda.hpp be included here?

Reply:
The previous approach was to include it in sycl.hpp, but I think this approach is more in the spirit of interop.

However, as for backend/opencl.hpp we need to include it by default if we want to let `get_mem` and `get_queue` on `interop_handler` default to opencl native objects, which allows for better backwards compatibility. I'm open to whether or not this should be done, so I think we have two possible paths for this:

1. Always include backend/opencl.hpp so we can use it as default for `get_mem` and `get_queue`.
2. Don't include backend/opencl.hpp and let the user include it if they need it. In this we could still let `get_mem` and `get_queue` default to use opencl for the native handles, but it would require the user to include backend/opencl.hpp themselves which is not transparent behavior.

Reply:
If I read the proposal correctly I think we should separate two things:

1. SYCL1.2.1 get() methods which return OpenCL handlers. For these we already include CL/cl.h where it's needed(hopefully) in the headers so no additional header is required to be included by a user.
2. get_native and other methods that generalization proposal adds. For these user needs to explicitly include backend/backend.hpp.

@Ruyk Please correct me if my understanding of the proposal is not correct.
https://github.com/KhronosGroup/SYCL-Shared/blob/master/proposals/sycl_generalization.md

Reply:
I agree that not including the backend headers and letting the user include it themselves when they know they need them is the better option and probably what was intended in the proposal.

However, the proposal doesn't mention the `interop_handler` so I've taken the liberty of generalizing it as well using the backend that has been added, but to avoid this breaking user code we have to default the backend for `get_mem` and `get_queue` to OpenCL. This means that in order to determine the default case of these methods we must include the backend header defining the OpenCL native types or require that the user does it (even though they don't use `backend::opencl` directly).